step/src/main/java/com/google/sps/servlets/PrefServlet.java

```diff
@@ -31,13 +31,33 @@
 @WebServlet("/Pref")
 public class PrefServlet extends HttpServlet {
   // The class stores the user' personal information data
+  private static final String UNKNOWN = "Unknown";
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String q1 = request.getParameter("q1");
+    if (q1 == null) {
+      q1 = UNKNOWN;
+    }
+
```

Comment:
Have you considered using ternary operators here? I think it would save a lot of lines of code but it might not look great so it's up to you.
String q1 = request.getParameter("q1");
q1 = q1 == null ? UNKNOWN : q1;
or
String q1 = request.getParameter("q1") == null ? UNKNOWN : request.getParameter("q1");